### PR TITLE
Add experimentation service

### PR DESCRIPTION
**Pull request by RMacfarlane**
Adopt the `vscode-tas-client` package for running experiments

### CODE REVIEW

File: src/experimentationService.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+		default: return TargetPopulation.Public;
+	}
+}
+
+export async function createExperimentationService(context: vscode.ExtensionContext, experimentationTelemetry: ExperimentationTelemetry) {
+	const pkg = await getPackageConfig(context.extensionPath);
+	const product = await getProductConfig(vscode.env.appRoot);
+	const targetPopulation = getTargetPopulation(product);
+
+	return getExperimentationService(`${pkg.publisher}.${pkg.name}`, pkg.version, targetPopulation, experimentationTelemetry, context.globalState);
```
Comment (RMacfarlane) [inline src/experimentationService.ts:83]: I'm not sure what the consequence the insiders and stable versions of the extension having different names here is, should they map to the same service?

Comment (sana-ajani) [reply src/experimentationService.ts:83]: cool, we confirmed with Julie and Amy about showing this only in the stable version of the extension

**Comment by sana-ajani** [comment]
Looks good for the most part! Only other thing I think we need is to add the GDPR classification for the "query-expfeature" telemetry event (like we do in the remote-containers extension: https://github.com/microsoft/vscode-remote-containers/blob/9c5f61a5ff6c390e148c237ab6b0e110f204f015/src/utils/telemetry.ts#L9).

**Comment by RMacfarlane** [comment]
Thanks! Added that classification

**Pull request merged**